strawberry/types/type_resolver.py

```diff
@@ -385,7 +313,8 @@ def _get_fields(cls: Type) -> List[FieldDefinition]:
                 default_value=getattr(cls, field.name, undefined),
             )
 
-        field_name = cast(str, field_definition.origin_name)
+        field_name = field_definition.origin_name
+        assert field_name is not None  # Not Optional by this point
         field_definitions[field_name] = field_definition
```

Comment:
I think an assert is better than a cast, because if `field_name` really _is_ `None`, then we probably want to know. Mypy also knows this makes the variable know longer `Optional` so I see this as a win-win.